lib/api.rb

```diff
@@ -0,0 +1,317 @@
+    return result
+  end
+
+  # miner_threads - Returns or sets the number of CPU threads for the miner.
+  # - @param [Integer] set_threads (optional) - The number of threads to allocate for mining.
+  def miner_threads(set_threads = nil)
+    if set_threads
+      result = @rpc.request("minerThreads", set_threads)
+    else
+      result = @rpc.request("minerThreads")
+    end
+    return result
+  end
+
+  # min_fee_per_byte - Returns or sets the minimum fee per byte.
+  # - @param [Integer] set_min_fee (optional) - The new minimum fee per byte.
+  def min_fee_per_byte(set_min_fee = nil)
+    if set_min_fee
+      result = @rpc.request("minFeePerByte", set_min_fee)
+    else
+      result = @rpc.request("minFeePerByte")
+    end
+    return result
+  end
+
+  # mining - Returns `true` if client is actively mining new blocks.
+  def mining
+    result = @rpc.request("mining")
+    return result
+  end
+
+  # peer_count - Returns number of peers currently connected to the client.
+  def peer_count
+    result = @rpc.request("peerCount")
+    return result
+  end
+
+  # peer_list - Returns list of peers known to the client.
+  def peer_list
+    result = @rpc.request("peerList")
+    return result
+  end
+
+  # peer_state - Returns the state of the peer.
+  # - @param [String] peer_address - The address of the peer.
+  def peer_state(peer_address)
+    result = @rpc.request("peerState", peer_address)
+    return result
+  end
+
+  # pool - Returns or sets the mining pool.
+  # When no parameter is given, it returns the current mining pool. When a value is given as parameter, it sets the mining pool to that value.
+  # - @param [String/Boolean] pool_address (optional) - The mining pool connection string (url:port) or boolean to enable/disable pool mining.
+  def pool(pool_address = nil)
+    if pool_address
+      result = @rpc.request("pool", pool_address)
+    else
+      result = @rpc.request("pool")
+    end
+    return result
+  end
+
+  # pool_confirmed_balance - Returns the confirmed mining pool balance.
+  def pool_confirmed_balance
+    result = @rpc.request("poolConfirmedBalance")
+    return result
+  end
+
+  # pool_connection_state - Returns the connection state to mining pool.
+  def pool_connection_state
+    result = @rpc.request("poolConnectionState")
+    return result
+  end
+
+  # send_raw_transaction - Sends a signed message call transaction or a contract creation, if the data field contains code.
+  # - @param [String] signed_transaction - The hex encoded signed transaction.
+  def send_raw_transaction(signed_transaction)
+    result = @rpc.request("sendRawTransaction", signed_transaction)
+    return result
+  end
+
+  # send_transaction - Creates new message call transaction or a contract creation, if the data field contains code.
+  # - @param [TransactionObject] transaction - The transaction object.
+  def send_transaction(transaction)
+    result = @rpc.request("sendTransaction", transaction)
+    return result
```

Comment:
Checking with the other RPC client implementations, could you please also include definitions for the returned objects and parameters where it makes sense. For reference, [the Java implementation](https://github.com/nimiq-community/java-client/tree/master/src/main/java/com/nimiq) is very complete version. The goal is to make it as easy as possible for the devs to use your lib and by having some form of struct/record/class/type (whatever it's called in Ruby :D) it will allow them to take advantage of code completion in their editors and IDEs.

Reply:
Hey! yea i know what you mean 😄, already thought of that before now and did something about it. The thing is  Ruby is a dynamically typed language so types cannot be pre defined and the variables changes its own type when the type of a newly assigned data has changed. That's why i prepared the tests to automatically do checks based on the returning results if it should be objects, numbers, strings and so on.

Example, here an account balance must be returned in a number shape, so Numeric :)
```ruby
it "Must be able to retrieve balance of account." do
    expect(@nimiq.get_balance(@main_addr)).to be_a_kind_of(Numeric)
end
```

Reply:
Remembered a third party library that stripe created, it helps with the data types for ruby. (non official)
Well thats just ruby luxury :D. Added it with full coverage, i'll be pushing soon!